File: code/C/Player.c
#include "Player.h"

#include <stdlib.h>
#include <string.h>

char* getName(Player* player) {
    return player->name;
}

int getLowestGuesses(Player* player) {
    return player->lowestGuesses;
}

int getHighestScore(Player* player) {
    return player->highestScore;
}

int getScore(Player* player) {
    return player->score;
}

void setScore(Player* player, int score) {
    player->score = score;
}

void setLowestGuesses(Player* player, int lowestGuesses) {
    player->lowestGuesses = lowestGuesses;
}

void setHighestScore(Player* player, int highestScore) {
    player->highestScore = highestScore;
}

int compareHighestScore(const void* a, const void* b) {
    Player* playerA = *(Player**)a;
    Player* playerB = *(Player**)b;
    return playerB->highestScore - playerA->highestScore;
}

int compareLowestGuesses(const void* a, const void* b) {
    Player* playerA = (Player*)a;
    Player* playerB = (Player*)b;
    return playerA->lowestGuesses - playerB->lowestGuesses;
}

File: code/Java/Player.java
public class Player {
    private String name;
    private int lowestGuesses;
    private int highestScore;
    private int score;

    public Player(String name, int lowestGuesses) {
        this.name = name;
        this.lowestGuesses = lowestGuesses;
    }

    public String getName() {
        return name;
    }

    public int getLowestGuesses() {
        return lowestGuesses;
    }
    
    public int getHighestScore() {
        return highestScore;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.lowestGuesses = score;
    }

    public void setLowestGuesses(int lowestGuesses) {
        this.lowestGuesses = lowestGuesses;
    }

    public void setHighestScore(int highestScore) {
        this.highestScore = highestScore;
    }
}
